huaweicloud/services/live/resource_huaweicloud_live_transcoding.go

```diff
@@ -0,0 +1,313 @@
+package live
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/live/v1/model"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+const ()
+
+func ResourceTranscoding() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceTranscodingCreate,
+		UpdateContext: resourceTranscodingUpdate,
+		DeleteContext: resourceTranscodingDelete,
+		ReadContext:   resourceTranscodingRead,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"domain_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"app_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"video_encoding": { //codec
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"H264", "H265"}, false)),
+			},
+
+			"templates": {
+				Type:     schema.TypeList,
+				Required: true,
+
+				MinItems: 1,
+				MaxItems: 4,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringMatch(regexp.MustCompile("[a-zA-Z0-9-]{1,64}$"),
+								"The name can contain a maximum of 64 characters, and only contains letters,"+
+									" digits and hyphens (-)."),
+						},
+
+						"width": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+
+						"height": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+
+						"bitrate": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(40, 30000),
+						},
+
+						"frame_rate": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(0, 30),
+							Computed:     true,
+						},
+					},
+				},
+			},
+
+			"low_bitrate_hd": { //hdlb
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "off",
```

Comment:
seems the default value of this is off on API side, we don't need to set this Default here again

Reply:
ok,done